### PR TITLE
Allow defining Digital Mic pins in config file, and add config for M5Stack Atom Lite

### PR DESCRIPTION
**Pull request by embedded-creations**

### CODE REVIEW

**Comment by embedded-creations** [comment]
Note that this env uses `platform_packages` which is define on master but not on dev, so that line should probably be taken out before merging

File: platformio.ini [hunk 1]
```diff
@@ -499,6 +499,25 @@ platform = ${common.platform_wled_default}
 board_build.ldscript = ${common.ldscript_2m512k}
 build_flags = ${common.build_flags_esp8266} -D LEDPIN=3 -D BTNPIN=1
 
+[env:esp32_m5stack_atom_lite]
+board = pico32
+platform = espressif32@2.0
+platform_packages = ${esp32.platform_packages}
+build_unflags = ${common.build_unflags}
+build_flags = ${common.build_flags_esp32} -D WLED_DISABLE_ALEXA -D WLED_DISABLE_BLYNK -D WLED_DISABLE_CRONIXIE
+  -D WLED_DISABLE_HUESYNC -D WLED_DISABLE_INFRARED -D WLED_DISABLE_MQTT
```
Comment (epsimatic) [inline platformio.ini:510]: Random WLED user here: 

I understand why you are disabling Alexa, Hue, MQTT, etc (I did the same with services I don't use), but it shouldn't be hardware specific and it shouldn't be in master branch.

Comment (embedded-creations) [reply platformio.ini:510]: Good point, I must have copied my config from an ESP8266 config that had those things disabled.  I'm just going to delete this PR anyway, as the important part (I2S pin redefinition support) I already committed to the dev branch

**Comment by embedded-creations** [comment]
Withdrawing as my config has some issues, no one seems particularly interested in the Atom Lite, and the "Allow defining Digital Microphone pins in config file" change has already been committed to `dev`

**Comment by epsimatic** [comment]
> no one seems particularly interested in the Atom Lite

I'm using it everywhere

`env:m5atom` is already supported in WLED, you only need to redefine some pins if you use custom hardware

**Comment by embedded-creations** [comment]
Good to know.  Have you noticed the 3.3V output from the Atom Lite is extremely noisy?  It makes the output from an analog or digital microphone unusably noisy, even with external cap added.  I needed to add an external 5V->3.3V regulator to work with a microphone.  I haven't tried powering any other peripherals with the 3.3V pin yet

**Comment by embedded-creations** [comment]
Also, do you have any current way to make the onboard WS2812 LED - or onboard WS2812 matrix on the Atom Lite Matrix - work when driving external LEDs?

**Comment by embedded-creations** [comment]
@epsimatic I probably should have tagged you with my questions before.  Tagging you now:

> Have you noticed the 3.3V output from the Atom Lite is extremely noisy? It makes the output from an analog or digital microphone unusably noisy, even with external cap added. I needed to add an external 5V->3.3V regulator to work with a microphone. I haven't tried powering any other peripherals with the 3.3V pin yet

> Also, do you have any current way to make the onboard WS2812 LED - or onboard WS2812 matrix on the Atom Lite Matrix - work when driving external LEDs?

(I since learned about the multichannel output effort going on for WLED, maybe there will be an easy solution to drive the onboard WS2812(s) then

**Comment by epsimatic** [comment]
> Have you noticed the 3.3V output from the Atom Lite is extremely noisy? 

Never used that, sorry. 

> It makes the output from an analog or digital microphone unusably noisy, even with external cap added. I needed to add an external 5V->3.3V regulator to work with a microphone.

Sorry, never used that **but** I've used m5atom Echo with its internal mic, and that works great after some i²s mic tweaks. 

I'll send a MR with new `env:m5echo` with tweaks after I figure out what exactly is needed

IMG_2959

> Also, do you have any current way to make the onboard WS2812 LED - or onboard WS2812 matrix on the Atom Lite Matrix - work when driving external LEDs?

I had that by accident: I've uploaded a generic m5atom build (which drives internal LED), rebooted, then uploaded a customised build that drives Grove connector yellow wire, then until reboot I had both showing the same output.

It's definitely a bug in WLED or firmware uploader, but it was nice.

Easy solution would be to bridge int LED and ext LEDs together, with either hardware bridge (and that might be really hard to do) or software. Easiest software solution would be to **duplicate** output on both pins. That sounds like a custom module, and that should be sent to upstream (Aircookie's) version

**Comment by embedded-creations** [comment]
@epsimatic that's great, there's at least a couple people interested in using the PDM mic that's in the M5stickC, Atom Echo, and available as a Grove module externally.

**Pull request closed without merging**